Integrations/integration-RsaNetWitnessPacketsAndLogs.yml

```diff
@@ -49,24 +49,115 @@ script:
         return url;
     };
 
+    var isObjectEmpty = function (obj) {
+        return Object.keys(obj).length === 0;
+    }
+
+    var mapTypeToValue = function (fields) {
+        var result = {};
+        var TYPE = 'type';
+        var VALUE = 'value';
+        if(fields && Array.isArray(fields)) {
+            fields.forEach(function (element) {
+                if (element.hasOwnProperty(TYPE) && element[TYPE]) {
+                    if(!Array.isArray(result[element[TYPE]])) {
+                        result[element[TYPE]] = [];
+                    }
+                    if(element.hasOwnProperty(VALUE) && element[VALUE]) {
+                        result[element[TYPE]].push(element[VALUE]);
+                    }
+                }
+            });
+        }
+        return result;
+    }
+
+    var buildHrAndContext = function (data) {
+        //Check if we have values
+        var retVal = {md:[], context: {Netwitness:{Events:[]}}};
+        var typeToValue;
+
+        if((typeof data) ==='object' && Array.isArray(data)) {
```

Comment:
Would create an inner function to handle a single case, and call it instead of duplicating the code
see ThreatGrid as ref